### PR TITLE
Add single date picker

### PR DESCRIPTION
**Pull request by mchaver**
- Add SingleDatePicker
- Add example for SingleDatePicker.
- Use webpack for example.

I mostly copied what you have for DateRangePicker. I still haven't figured out how to clear the focus after selecting a date though. Any ideas?

### CODE REVIEW

**Comment by liubko** [comment]
@mchaver thanks for PR. Why do you decide to eject react-scripts in examples folder?

**Comment by liubko** [comment]
maybe you wouldn't use it in a real project, but I think it suits well here in a demo examples folder.

**Comment by mchaver** [comment]
@liubko I wasn't seeing the output, but maybe I was going to the wrong port? I am fine with reverting it back to react-scripts.

**Comment by liubko** [comment]
maybe you can try it again? I think it should open on 8080 if I recall correctly, I can check this a bit later.

**Comment by mchaver** [comment]
I was able to get it working with react-scripts. I removed the webpack stuff. Do you have any ideas about the focus issue? Maybe you can try it out locally

**Comment by liubko** [comment]
Here is the difference between `DateRangePicker` and `SingleDatePicker`,
in `SingleDatepicker` `onFocusChange` receives an object of shape `{focused: bool}`
https://github.com/airbnb/react-dates#singledatepicker

- First, you need to update the binding
```
[@bs.obj] external makeProps :
  (
    ...
    ~onFocusChange: ({. "focused": bool}) => unit,
    ...
  )
```
- Second, you need to fix the example
```
type action =
   ...
  | FocusChange(bool);

... 

type state = {
  ...
  focused: bool,
};

...

  reducer: (action, state) =>
    switch (action) {
    ...
    | FocusChange(v) => ReasonReact.Update({...state, focused: v})
    },

...

  render: self => {
    <SingleDatePicker
      ...
      onFocusChange=(v => {
        self.send(FocusChange(v##focused));
      })
    />;
```

- please remove all unrelated changes from PR. `(examples/dist, yarn.lock, etc)`

**Comment by mchaver** [comment]
@liubko Thanks! I think I have everything.

File: src/BsReactDates__SingleDatePicker.re [hunk 1]
```diff
@@ -0,0 +1,214 @@
+open BsReactDates__Utils;
+
+open MomentRe;
+
+let nullableFocusedInputToJs = v =>
```
Comment (liubko) [inline src/BsReactDates__SingleDatePicker.re]: this isn't used anywhere

**Comment by liubko** [comment]
@mchaver One thing that can be improved, we can provide user with a better API, we can convert `onFocusChange: {. "focused": bool} => unit` to `onFocusChange: bool => unit`.

So the usage will be 
```
<SingleDatePicker
   ...
   onFocusChange=(focused => {
      self.send(FocusChange(focused));
   )
/>
```

Let me know if you interested in doing this, otherwise I can change it myself later.

**Comment by mchaver** [comment]
sure i can make those changes today

**Comment by jcleefw** [comment]
this is what i'm looking for. Any idea on whether we can get this PR merge in?

**Comment by jcleefw** [comment]
this is what i'm looking for. Any idea on whether we can get this PR merge in?

**Comment by jcleefw** [comment]
this is what i'm looking for. Any idea on whether we can get this PR merge in?

**Comment by liubko** [comment]
@jcleefw maybe you want to take over this part? https://github.com/ahrefs/bs-react-dates/pull/1#issuecomment-402027344 this will speed up things

**Comment by liubko** [comment]
merged as a part of https://github.com/ahrefs/bs-react-dates/pull/2

**Pull request closed without merging**